src/app/clusters/ota-requestor/DefaultOTARequestor.h

```diff
@@ -147,6 +147,12 @@ class DefaultOTARequestor : public OTARequestorInterface, public BDXDownloader::
                 Reset();
             }
 
+            if (!sendFlags.Has(chip::Messaging::SendMessageFlags::kExpectResponse))
```

Comment:
This also seems unrelated to polling per se and should probably be a separate PR.

Reply:
I think this causes a crash on the requestor side when testing with the app and my code changes. Its been a while I tested this. So i am going to try again. I will update the PR comment with details.